spring-integration-redis/src/test/java/org/springframework/integration/redis/rules/RedisAvailableTests.java

```diff
@@ -85,6 +92,26 @@ protected void awaitContainerSubscribedWithPatterns(RedisMessageListenerContaine
 		Thread.sleep(1000);
 	}
 
+	protected void awaitFullySubscribed(RedisMessageListenerContainer container, RedisTemplate<?, ?> redisTemplate,
+			String redisChannelName, QueueChannel channel, Object message) throws Exception {
+		awaitContainerSubscribedNoWait(container);
+		drain(channel);
+		long now = System.currentTimeMillis();
+		Message<?> received = null;
+		while (received == null && System.currentTimeMillis() - now < 10000) {
+			redisTemplate.convertAndSend(redisChannelName, message);
+			received = channel.receive(1000);
+		}
+		drain(channel);
+		assertNotNull("Container failed to fully start", received);
+	}
+
+	private void drain(QueueChannel channel) {
+		while (channel.receive(0) != null) {
+			;
+		}
```

Comment:
I would say no if this was production code (construct, build and discard unused collection) but since this is a test, yes.